planner/core/plan_cost.go

```diff
@@ -300,21 +307,27 @@ func (p *PhysicalTableReader) GetPlanCost(_ property.TaskType, option *PlanCostO
 	}
 	p.planCost = 0
 	netFactor := getTableNetFactor(p.tablePlan)
-	switch p.StoreType {
+	var rowCount, rowSize, netSeekCost, tableCost float64
+	sqlScanConcurrency := p.ctx.GetSessionVars().DistSQLScanConcurrency()
+	storeType := p.StoreType
+	switch storeType {
 	case kv.TiKV:
 		// child's cost
 		childCost, err := p.tablePlan.GetPlanCost(property.CopSingleReadTaskType, option)
 		if err != nil {
 			return 0, err
 		}
+		tableCost = childCost
 		p.planCost = childCost
 		// net I/O cost: rows * row-size * net-factor
-		rowSize := getTblStats(p.tablePlan).GetAvgRowSize(p.ctx, p.tablePlan.Schema().Columns, false, false)
-		p.planCost += getCardinality(p.tablePlan, costFlag) * rowSize * netFactor
+		rowSize = getTblStats(p.tablePlan).GetAvgRowSize(p.ctx, p.tablePlan.Schema().Columns, false, false)
+		rowCount = getCardinality(p.tablePlan, costFlag)
+		p.planCost += rowCount * rowSize * netFactor
 		// net seek cost
-		p.planCost += estimateNetSeekCost(p.tablePlan)
+		netSeekCost = estimateNetSeekCost(p.tablePlan)
+		p.planCost += netSeekCost
 		// consider concurrency
-		p.planCost /= float64(p.ctx.GetSessionVars().DistSQLScanConcurrency())
+		p.planCost /= float64(sqlScanConcurrency)
 	case kv.TiFlash:
```

Comment:
The `kv.TiFlash` case is not handled.

Reply:
Optimize Tracer can't handle tiflash well for now, thus we don't cover tiflash currently.